W-14476576 DC sync not req duke (includes W-14476577,  W-14476579,  W-14776338) (#263)
mulesoft/docs-code-builder: modules/ROOT/pages/start-scm.adoc

//open ACB terminal:
include::anypoint-code-builder::partial$acb-reusable-steps.adoc[tag="open-terminal"]
. Navigate to your home directory for Anypoint Code Builder.
// Pointer to Command Palette
include::partial$acb-reusable-steps.adoc[tags="open-command-palette"]
. Clone the repo into the Source Control panel for your project:
.. Provide the following command to add the remote repo on GitHub:
+
[source, command]
----
Git: Clone
----
.. Provide the URL to your repo in GitHub, for example:
+
[source,command]
----
https://github.com/_owner_/_yourrepo_.git
----
+
Replace the URL in the example with the URL to your repo in GitHub. The owner and repo URL are unique to your GitHub site.
.. Navigate to a directory for your new repo, and click *Select as repo Destination*.
.. When prompted, click *Open* to open the project.

// Link to sync-changes
include::anypoint-code-builder::partial$acb-reusable-steps.adoc[tags="scm-sync-changes"]

[[check-remote]]
== Get the URL of a Project's Remote Repository in GitHub

> To get for the remote repository's URL:

Author: ```suggestion
To get the remote repository's URL:
```

If you are unsure whether your project is already connected to a remote repository, see <<check-remote>>.

This example assumes that you have an empty repo in GitHub. If you do not, you can create one from GitHub. For guidance, see https://docs.github.com/en/repositories/creating-and-managing-repositories/quickstart-for-repositories[Quickstart for repositories^] in the GitHub documentation.  

. Meet the prerequisites in <<prereqs>>.
//step: open the project folder
include::anypoint-code-builder::partial$acb-open-packaged-sources.adoc[tag="open-workspace-root"]
// Pointer to Source Control Panel
include::partial$acb-reusable-steps.adoc[tags="open-source-control"]
.. In the Source Control Panel, notice that there is a repo in the *Source Control Repositories* area of the panel, for example:
+
image::start-scm-oas-anypoint.png["Project source control using the Anypoint SCM"]
// Pointer to Command Palette
include::partial$acb-reusable-steps.adoc[tags="open-command-palette"]
. Configure a connection from the project in VS Code to your repo in GitHub:
.. Provide the following command to add the remote repo on GitHub:
+
[source, command]
----

> Git: Add Remote...

Reviewer: Same question as above. I'm not sure if this is the command or the search text, and I'm not sure if they should enter the ellipses into the search field. I'll test later if I can.

----
.. Select *Add Remote from GitHub* from the popup menu.
+
image::start-scm-oas-add-remote.png["Adding an extra remote repo from GitHub"]
.. Provide the URL to your git repo, for example:
+
----
https://github.com/me/my-extra-repo.git

> ----

Reviewer: Should this snippet (and the following snippet) have attributes that make it able to be copied into the clipboard? It might be good to do so if customers need to enter this text. Again though, I'm not yet testing the actual UI as I review, so maybe this is a non-issue.

Author: I don't think they should copy this example because it won't work for them. I'm trying to help avoid someone making a mindless mistake here.

.. Provide a name for the remote in VS Code, for example:
+
----
my-extra-repo
----
+
This name for the repo in VS Code can match or differ from the repo name in GitHub.
. Publish your project to your extra remote repo:
.. In the command palette, provide the following command:
+
[source, command]
----

> Git: Push to...

Reviewer: See question about having to enter the ellipses. Probably a non-issue, but I thought I'd check.

Author: same thing. It's what they type into the IDE's command palette, not in the command terminal.

Author: Maybe a graphic would be better here? If they're following the procedure though, it should be obvious. We use this style everywhere in the ACB docs. And this is how we document use of the command palette.

Author: I'm leaning against changes.

:repo-in-github: If you have an empty repo in GitHub, you can publish your project files to the repo from VS Code.
:extra-repo: If your project is already configured to sync with a remote repo, you can configure VS Code to publish your project files to a different repo. 
:extra-repo-continued: This feature is helpful, for example, if you want to use your own repo for an API spec project that you imported from Design Center or if you created your design project from the IDE before the February 2024 release. Both types of projects automatically add a connection to a remote Git repo in the Anypoint SCM.

include::reuse::partial$beta-banner.adoc[tag="anypoint-code-builder"] 

// Shared intro to SCM
include::anypoint-code-builder::partial$acb-reusable-steps.adoc[tag="scm-intro-best-practice"]
A useful SCM provides change history of each file and a branching methodology to maintain project quality, full traceability, the ability to restore earlier versions of your files, and a way to recover source code if your Anypoint Code Builder environment is lost or removed. 

> GitHub and the Git command line interface (CLI) are widely used in the industry and supported by VS Code out-of-the-box, with minimal configuration. The procedures on this page show how to set up and use GitHub for source control. VS Code extensions for other SCM systems are also available for you to set up and use. For information, see https://code.visualstudio.com/docs/sourcecontrol/overview#_scm-providers[SCM Providers^]. 

Reviewer: Should a mention of cloud IDE be made here as well? I'm thinking of the scenario where a customer skips the overview and then wonders if the cloud IDE supports SCM. I mean, nobody should be skipping anything, but we know how it is! ;)

Author: ```suggestion
GitHub and the Git command line interface (CLI) are widely used in the industry and supported by VS Code out-of-the-box, with minimal configuration, in the desktop and cloud IDE. The procedures on this page show how to set up and use GitHub for source control. VS Code extensions for other SCM systems are also available for you to set up and use. For information, see https://code.visualstudio.com/docs/sourcecontrol/overview#_scm-providers[SCM Providers^]. 
```

Author: The cloud IDE is an implementation of VS Code called code-server.

Author: I tweaked it a bit, but support for SCM extensions is a VS Code feature supported within the cloud and desktop IDEs because they are both VS Code. We cannot suggest that this is our feature.

[[prereqs]]
== Before You Begin

Ensure that these prerequisites are in place:

* A GitHub account
+
If you do not have an account, create one from the https://github.com/join[Join GitHub^] page on the GitHub site.

* A Git installation on your computer 
+
If Git is not installed on your computer, follow the https://github.com/git-guides/install-git[Git installation instructions^] for your machine.

* Access to Anypoint Code Builder from the desktop or cloud IDE
+
For setup and installation information, see xref:start-acb.adoc[], and be sure to meet all prerequisites. 

* A GitHub user name and email configuration (see <<config-git>>)

[[config-git]]
=== Configure Your Git Username and Password

Check for and configure your username and email properties so that you can add, update, and retrieve your project files from a remote Git repository (repo) in GitHub. 

. Meet the prerequisites in <<prereqs>>.
//open ACB terminal:
include::anypoint-code-builder::partial$acb-reusable-steps.adoc[tag="open-terminal"]

> . From the Anypoint Code Builder terminal, check for any existing Git configuration by running this command:

Reviewer: On Desktop, I assume they can use any terminal client. Is there a specific reason why we want them to use the ACB terminal? If so, should this be mentioned? Or are we simply consolidating cloud/desktop IDE steps here?

Author: It works in cloud and desktop IDE if they use the IDE's terminal instead of their own. They can use their own terminal in desktop. I don't think we need to go into detail about terminals here though.

* <<create-repo>>
+
//shared info about this procedure:
{no-repo-yet}
* <<initialize-repo>>
+
// shared info about initialization
{init-local}
* <<publish-to-empty>>
+
//shared info about this procedure:
{repo-in-github}
//shared info about this procedure:
{extra-repo}
//shared description:
{extra-repo-continued}
* <<clone-repo>>
+
If a design, implementation, or integration project is already stored in a repo on GitHub, you can clone it into the cloud or desktop IDE. 

[[create-repo]]
=== Generate and Publish to a Repository in GitHub

//shared info about this procedure:
{no-repo-yet} 
If you already have an empty repo in GitHub that you intend to use for a project in Anypoint Code Builder, see <<publish-to-empty>> to add and publish to the repo from VS Code. 

. Meet the prerequisites in <<prereqs>>.
//step: open the project folder
include::anypoint-code-builder::partial$acb-open-packaged-sources.adoc[tag="open-workspace-root"]
// Pointer to Source Control Panel
include::partial$acb-reusable-steps.adoc[tags="open-source-control-no-edits"]
. In the Source Control panel, click *Publish to GitHub*.
+
image::start-scm-repo.png["Publish repo to GitHub from VS Code"]
+

> When no repo is associated with your project, VS Code provides options for initializing a _local_ repo (*Initialize Repository*) on your IDE or directly publishing your project to a new remote repo in GitHub (*Publish to GitHub*).

Reviewer: I wonder if this is confusing. Publishing directly to a new GitHub repo also implicitly initializes a local repo in the currently open directory. Maybe not mention "local" repo? Maybe it's more an initialization of git in this currently open directory?

Author: I wasn't clear at first that the initialization was for a local repo vs. the remote. So the goal was to make that clear. I can add they that it implicitly initialized the local repo though. What could be confusing, I think, is that the repository you see in VS Code UI is the local one.

. Select a repo type (private or public).
+
To protect data in your repo, select a private repo (*Publish to GitHub private repo*). This process commits and pushes your project files to your new repo.
. Select the files to publish to the repo.
+
You can select all or a subset of the files.
. To view your project files in a GitHub repo, click *Open on GitHub* in the popup. 
+
image::start-scm-repo-open-in-github.png["Opening repo in GitHub"]
+
Alternatively, open GitHub from your browser and locate your new repo.

// Link to sync-changes
include::anypoint-code-builder::partial$acb-reusable-steps.adoc[tags="scm-sync-changes"]

[[initialize-repo]]
=== Initialize and Commit Project Files to a Local Repository

// shared info about initialization
{init-local}

. Meet the prerequisites in <<prereqs>>.
//step: open the project folder
include::anypoint-code-builder::partial$acb-open-packaged-sources.adoc[tag="open-workspace-root"]
// Pointer to Source Control Panel
include::partial$acb-reusable-steps.adoc[tags="open-source-control-no-edits"]
. From the Source Control panel for your project, initialize and commit your project files to a local repository in the IDE for your project:

> .. In the Source Control panel, click *Initialize Repository* to generated a local repository in your IDE.

Reviewer: ```suggestion
.. In the Source Control panel, click *Initialize Repository* to generate a local repository for your project.
```
See note above.

* <<create-repo>>
+
//shared info about this procedure:
{no-repo-yet}
* <<initialize-repo>>
+
// shared info about initialization
{init-local}
* <<publish-to-empty>>
+
//shared info about this procedure:
{repo-in-github}
//shared info about this procedure:
{extra-repo}
//shared description:
{extra-repo-continued}
* <<clone-repo>>
+
If a design, implementation, or integration project is already stored in a repo on GitHub, you can clone it into the cloud or desktop IDE. 

[[create-repo]]
=== Generate and Publish to a Repository in GitHub

//shared info about this procedure:
{no-repo-yet} 
If you already have an empty repo in GitHub that you intend to use for a project in Anypoint Code Builder, see <<publish-to-empty>> to add and publish to the repo from VS Code. 

. Meet the prerequisites in <<prereqs>>.
//step: open the project folder
include::anypoint-code-builder::partial$acb-open-packaged-sources.adoc[tag="open-workspace-root"]
// Pointer to Source Control Panel
include::partial$acb-reusable-steps.adoc[tags="open-source-control-no-edits"]
. In the Source Control panel, click *Publish to GitHub*.
+
image::start-scm-repo.png["Publish repo to GitHub from VS Code"]
+

> When no repo is associated with your project, VS Code provides options for initializing a _local_ repo (*Initialize Repository*) on your IDE or directly publishing your project to a new remote repo in GitHub (*Publish to GitHub*).

Author: ```suggestion
When no repo is associated with your project, VS Code provides options for initializing a _local_ repo (*Initialize Repository*) on your IDE or directly publishing your project to a new remote repo in GitHub (*Publish to GitHub*). Publishing to GitHub also implicitly initializes your local repo for the project. 
```

. Select the files to publish to the repo.
+
You can select all or a subset of the files.
. To view your project files in a GitHub repo, click *Open on GitHub* in the popup. 
+
image::start-scm-repo-open-in-github.png["Opening repo in GitHub"]
+
Alternatively, open GitHub from your browser and locate your new repo.

// Link to sync-changes
include::anypoint-code-builder::partial$acb-reusable-steps.adoc[tags="scm-sync-changes"]

[[initialize-repo]]
=== Initialize and Commit Project Files to a Local Repository

// shared info about initialization
{init-local}

. Meet the prerequisites in <<prereqs>>.
//step: open the project folder
include::anypoint-code-builder::partial$acb-open-packaged-sources.adoc[tag="open-workspace-root"]
// Pointer to Source Control Panel
include::partial$acb-reusable-steps.adoc[tags="open-source-control-no-edits"]
. From the Source Control panel for your project, initialize and commit your project files to a local repository in the IDE for your project:
.. In the Source Control panel, click *Initialize Repository* to generate a local repository for your project.
+

> When no repo is associated with your project, VS Code provides options for initializing a _local_ repo (*Initialize Repository*) on your IDE or directly publishing your project to a new remote repo in GitHub (*Publish to GitHub*).

Reviewer: ```suggestion
When no repo is associated with your project, VS Code provides options for initializing a _local_ repo (*Initialize Repository*) for your project or directly publishing your project to a new remote repo in GitHub (*Publish to GitHub*).
```

// Shared substep to stage files
include::partial$acb-reusable-steps.adoc[tags="scm-stage-files-substep"]
// Shared substeps to commit files
include::partial$acb-reusable-steps.adoc[tags="scm-commit-message-substep"]
+

> After committing, _do not click_ the button for publishing to a branch yet. Instead, you will use the command line to add a connection from your project to your remote repo on GitHub before publishing to a branch in that repo.

Reviewer: Perhaps spell out the literal button text to not click? Or is it clear enough from this description (I'm not able to test at the moment so I'm not seeing the actual UI flow.)

Author: I was worried about someone mindlessly clicking it if the label was written in big bold type, so I wanted them to either read the note or skip it.

To publish your files to a remote repository, you can:

* <<publish-to-empty>>
+
//shared description:
{repo-in-github}
* <<add-extra-repo>>
+
//shared description:
{extra-repo}

[[publish-to-empty]]
=== Publish Project Files into an Empty Repository in GitHub

//shared info about this procedure:
{repo-in-github}
This procedure assumes that the project files are ready to stage and publish to the repo and that no other repo is present in the Source Control panel for your project. 

If another repo, such as the Anypoint SCM, is present in the Source Control panel for your project, you can <<sync-changes, sync your changes with that repo>>, or you can <<add-extra-repo, publish your changes to a different repo>>.

. Meet the prerequisites in <<prereqs>>.
. If you have not yet initialized a local repository for your project in the IDE, see <<initialize-repo>>.
// Pointer to Command Palette
include::partial$acb-reusable-steps.adoc[tags="open-command-palette"]
. Configure a connection from the project in VS Code to your empty remote repo in GitHub:
.. Provide the following command to add the remote repo on GitHub:
+
[source, command]
----

> Git: Add Remote...

Reviewer: Is this the literal command text? Or is it a search phrase? On line 186 it mentions that it's a command, which means the customer might think this is the literal command text. If it's a search phrase, should this be clear? Do they enter the ellipsis into the command palette search field?

Author: That's the literal command text _in the IDE_, which is different from what you enter on the command line. They are entering this from the IDE in this procedure. Do you think it would be easier to understand if in bold?  

<img width="607" alt="Screenshot 2024-01-18 at 9 16 21 AM" src="https://github.com/mulesoft/docs-code-builder/assets/10456406/1cab5ed1-784a-418f-ae8e-20375536afed">

Author: Adding it as a command enables them to easily copy and paste it into the IDE from the docs. As mentioned below, we do this all over the docs for that purpose.
